File: Week_4/OOPS/ToTestInt.java
interface Test {
    int square(int num);
}

class Arithmetic implements Test {
    @Override
    public int square(int num) {
        return num * num;
    }
}

public class ToTestInt {
    public static void main(String[] args) {
        Arithmetic arithmeticObj = new Arithmetic();
        int result = arithmeticObj.square(5);
        System.out.println("Square of 5 is: " + result);
    }
}

File: Week_4/OOPS/incpp/ToTestInt.cpp
#include <iostream>

class Test {
public:
    virtual int square(int num) = 0; 
    virtual ~Test() {} 
};

class Arithmetic : public Test {
public:
    int square(int num) override {
        return num * num;
    }
};

class ToTestInt {
public:
    void run() {
        Arithmetic arithmeticObj;
        int result = arithmeticObj.square(5);
        std::cout << "Square of 5 is: " << result << std::endl;
    }
};

int main() {
    ToTestInt testRunner;
    testRunner.run();
    return 0;
}
